sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/InMemoryCatalog.scala

```diff
@@ -245,7 +301,21 @@ class InMemoryCatalog extends ExternalCatalog {
           s"'$db' table '$table':\n$dupSpecsStr")
       }
     }
-    parts.foreach { p => existingParts.put(p.spec, p) }
+
+    val tableDir = new Path(catalog(db).db.locationUri, table)
+    // TODO: we should follow hive to roll back if one partition path failed to create.
+    parts.foreach { p =>
+      if (p.storage.locationUri.isEmpty) {
+        val partitionPath = p.spec.map { case (k, v) => s"$k=$v" }.mkString("/")
```

Comment:
One potential issue is that the partition spec is a map and is un-ordered. In one of my PR, we had to first compute the order based on partition column names. Lian had found this issue while reviewing my PR :-)